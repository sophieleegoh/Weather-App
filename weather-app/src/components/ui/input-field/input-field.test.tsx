import renderer from 'react-test-renderer';
import InputField from './index';

jest.mock('react-hook-form', () => ({
  ...jest.requireActual('react-hook-form'),
  useFormContext: () => ({
    handleSubmit: () => jest.fn(),
    control: {
      register: jest.fn(),
      unregister: jest.fn(),
      getFieldState: jest.fn(),
      _names: {
        array: new Set('test'),
        mount: new Set('test'),
        unMount: new Set('test'),
        watch: new Set('test'),
        focus: 'test',
        watchAll: false,
      },
      _subjects: {
        watch: jest.fn(),
        array: jest.fn(),
        state: jest.fn(),
      },
      _getWatch: jest.fn(),
      _formValues: ['test'],
      _defaultValues: ['test'],
    },
    getValues: () => {
      return [];
    },
    setValue: () => jest.fn(),
    formState: () => jest.fn(),
    watch: () => jest.fn(),
  }),
  Controller: () => [],
  useSubscribe: () => ({
    r: { current: { subject: { subscribe: () => jest.fn() } } },
  }),
}));

describe("Input field snapshot tests", () => {
  it("Renders the component", () => {
    const component = renderer.create(
      <InputField placeholder="placeholder text" inputText="" onUpdate={() => {}} />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
})