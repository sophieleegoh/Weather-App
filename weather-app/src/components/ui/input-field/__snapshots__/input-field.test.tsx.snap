// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field snapshot tests Renders the component 1`] = `
<input
  onChange={[Function]}
  placeholder="placeholder text"
  type="text"
  value=""
/>
`;
